### PR TITLE
Improve table formatting for ms teams integration

### PR DESCRIPTION
**Pull request by maggie-lin**
**Why**
Tables both aren't supported by the adaptive-html library and MS teams. Tables from Guru (left) turn into paragraphs (right) and look pretty bad right now. 

imageimage

**What**
Turn Guru html tables into adaptive card column set. If a Guru table has more than 3 columns or has any cells with more than 100 characters, a fallback message [as defined here](https://docs.google.com/spreadsheets/d/1U6vywuTWXtTbmAW86U3YP_qPCS6RGikNseQKiCBr_kI/edit#gid=0) will be surfaced for a customer to view the card in the app instead. 
https://www.loom.com/share/aecb6a5a1d5d4fd8a64513cb588568a3

### CODE REVIEW

File: dist/adaptive-html.cjs.js [hunk 1]
```diff
@@ -333,6 +357,67 @@ rules.image = {
...
+
+    if (columns > maxColumns) {
+      return createTextBlock(fallbackText);
+    }
+
+    for (var i = 0; i < rows; i++) {
+      var items = content[i].items || [];
+
+      if (items.some(function (item) {
+        return (item.text || '').length > maxCellCharacters;
```
Comment (austinparkk) [inline dist/adaptive-html.cjs.js:379]: is this saying if the length of  item.text or '' is greater than maxCellCharacters? isnt the length of '' never going to be greater than maxCellCharacters?

**Review by austinparkk — COMMENTED** [review]
seems like theres a lot of duplicate code. Not sure what the different files are for but is there a way around this?

**Comment by maggie-lin** [comment]
@austinparkk yeahh all the duplicates are prebuilt versions of the library in `dist/` - according the readme it's for browser script (iife), CommonJS module environments (cjs), ES module environments (es). 

I'm using the cjs lib in our app integration, but for compatibility reasons (like contributing back to the original repo) I'm continuing to update everything (using `npm run build`). The tests also run against the cjs lib as well, so we could decide to stop updating the others but and just update the readme

**Comment by austinparkk** [comment]
> @austinparkk yeahh all the duplicates are prebuilt versions of the library in `dist/` - according the readme it's for browser script (iife), CommonJS module environments (cjs), ES module environments (es).
> 
> I'm using the cjs lib in our app integration, but for compatibility reasons (like contributing back to the original repo) I'm continuing to update everything (using `npm run build`). The tests also run against the cjs lib as well, so we could decide to stop updating the others but and just update the readme

oh okay. Yea, im not really sure what is better so i will leave that up to you :) just wanted to bring it up just in case.

**Comment by maggie-lin** [comment]
@austinparkk yep, it's a good call! it's probably not necessary to have this level of duplication. but I'd definitely want some more input from FE's since I don't really know much about these types or why we'd want to keep them

File: dist/adaptive-html.cjs.js (hunk 1 above)
Comment (maggie-lin) [reply dist/adaptive-html.cjs.js:379]: this is checking the `maxCellCharacters` against text per table cell, which usually looks something like:
```
 {
    "type": "TextBlock",
    "text": "blah blah",
    "wrap": true
 }
```

so by the time the json gets to `rules.tableSection`, `content` looks something like the following as a result of how `rules.tableRow` and `rules.tableCell` are implemented:

```
     [
          {
              "type": "Container",
              "items": [
                  {
                      "type": "TextBlock",
                      "text": "blah blah",
                      "wrap": true
                  },
                  {
                      "type": "Image",
                      "url": "https://blah.com",
                      "altText": "image.jpg"
                  }
              ]
          },
          {
              "type": "Container",
              "items": [
                  {
                      "type": "TextBlock",
                      "text": "blah",
                      "wrap": true
                  },
                  {
                      "type": "TextBlock",
                      "text": "blah",
                      "wrap": true
                  }
              ]
          }
      ]
```

**Review by austinparkk — APPROVED** [review]

**Review by RamanjitKhakh — APPROVED** [review]

**Pull request merged**